.github/workflows/main.yml

```diff
@@ -216,8 +218,9 @@ jobs:
           pytest -s
 
       - name: Run Python tests with numpy<2.0 (ubuntu)
-        if: (matrix.python_version == '3.11' || matrix.python_version == '3.12' ) && (matrix.os == 'ubuntu-latest' || matrix.os == 'ubuntu-24.04')
+        if: (matrix.python_version == '3.11' || matrix.python_version == '3.12' ) && startsWith(matrix.os,'ubuntu')
         run: |
-          sudo apt-get install libjpeg-dev zlib1g-dev libpng-dev libtiff-dev libfreetype6-dev
+          # 2024.10.15: Error message: The headers or library files could not be found for jpeg, a required dependency when compiling Pillow from source.
+          sudo apt-get install libjpeg-dev zlib1g-dev libpng-dev
```

Comment:
Why do we need to compile pillow?

Reply:
The following runs fails: https://github.com/onnx/onnx/actions/runs/11306051777/job/31446447690
We want to have numpy < 2, which means pillow<10, so we have pillow 9.5.0 and that version is not supported by python 3.12 (https://pillow.readthedocs.io/en/latest/installation/python-support.html#python-support). Might work with python 3.11? I'm not sure why this was not a problem with ubuntu 22.04, was there a corresponding dependency already installed elsewhere?

Reply:
I guess we will want numpy2 support first then?

Reply:
I personally don’t think we should compile any dependencies

Reply:
> I guess we will want numpy2 support first then?

We have already numpy 2 support. Here at that step we make sure that everything also works with older numpy versions

Reply:
> I personally don’t think we should compile any dependencies

I'll have a look

Reply:
also our current pipeline with ubuntu-22.04 seems to build pillow: 
https://github.com/onnx/onnx/actions/runs/11352377878/job/31574997465#step:21:5845